src/fold/loop/encase.py

```diff
@@ -0,0 +1,39 @@
+from typing import TYPE_CHECKING, Tuple
+
+import pandas as pd
+from backtesting import backtest
+from splitters import Splitter
+from training import train
+from transformations.base import BlocksOrWrappable
+
+from ..all_types import OutOfSamplePredictions, TransformationsOverTime
+
+if TYPE_CHECKING:
+    from krisi import ScoreCard
```

Comment:
are we sure that this won't result in krisi being a hard dependency, for example, for development? does `TYPE_CHECKING` do anything with pylance, what we use?

Reply:
It will result in an error when in development. 
We can either: A.) Put it into a dev dependency or B.) Do this method without `scorecard` being type checked

Reply:
isn't scorecard always not type checked, as we wrap the type name into an `str`?

Reply:
Yes that is the behaviour.

<img width="538" alt="Screenshot 2023-03-23 at 18 49 57" src="https://user-images.githubusercontent.com/16992850/227298625-0dec87ea-42fa-4a20-bba1-cb5b946cede3.png">

Reply:
great!